File: app/src/main/java/com/ddu/ui/fragment/work/FragmentA.java
package com.ddu.ui.fragment.work;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.ddu.R;
import com.ddu.icore.rx.bus.RxBus;
import com.ddu.icore.ui.fragment.DefaultFragment;
import com.ddu.icore.ui.view.CustomerTimeLineMarker;
import com.ddu.icore.util.sys.ViewUtils;

import butterknife.OnClick;
import io.reactivex.functions.Consumer;

/**
 * Created by lhz on 16/4/6.
 */
public class FragmentA extends DefaultFragment {

    private Button mBtnOk;
    private Button button;

    private CustomerTimeLineMarker item_time_line_mark;


    @NonNull
    public static FragmentA newInstance() {
        FragmentA fragment = new FragmentA();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void initData(Bundle savedInstanceState) {


    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_work_state;
    }

    @Override
    public void initView() {
        mBtnOk = ViewUtils.findViewById(mView, R.id.btn_ok);
        button = ViewUtils.findViewById(mView, R.id.button);
        item_time_line_mark = ViewUtils.findViewById(mView, R.id.item_time_line_mark);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                RxBus.action("3").subscribe(new Consumer<Integer>() {
                    @Override
                    public void accept(Integer o) throws Exception {
                        Log.v("lhz", "hello: " + o);
                    }
                }, new Consumer<Throwable>() {
                    @Override
                    public void accept(Throwable throwable) throws Exception {
                        Log.v("lhz", "e: " + throwable.getMessage());
                    }
                });
//                ShareDialogFragment bottomSheetDialogFragment = new ShareDialogFragment();
//                bottomSheetDialogFragment.show(getFragmentManager(), "");
//                replaceFragment(FragmentB.newInstance());
//                ((ShowDetailActivity) mActivity).replaceFragment(FragmentB.newInstance(), FragmentUtils.FRAGMENT_ADD_TO_BACK_STACK);
            }
        });
        mBtnOk.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                RxBus.post("3", 1);
//                FragmentB f = FragmentB.newInstance();
//                replaceFragment(f);
//                item_time_line_mark.setCount(2);

            }
        });
        setDefaultTitle("FragmentA");
    }

    @OnClick(R.id.rl_person_info)
    public void onClick() {

    }
}

File: app/src/main/java/com/ddu/ui/fragment/work/FragmentA.kt
package com.ddu.ui.fragment.work

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import com.ddu.R
import com.ddu.icore.common.ext.ctx
import com.ddu.icore.ui.fragment.DefaultFragment
import com.ddu.ui.activity.TestActivity
import com.ddu.util.ToastUtils
import kotlinx.android.synthetic.main.fragment_work_state.*
import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
import java.net.URLEncoder

/**
 * Created by lhz on 16/4/6.
 */
class FragmentA : DefaultFragment() {

    private var schemeUrl: String? = null
        set(value) {
            startTo(value)
        }
    var yzbzzUrl: String? = null
    var httpUrl: String? = null
    var defaultData = "yzbzz://5"
    var login = "1"

    override fun initData(savedInstanceState: Bundle?) {
        yzbzzUrl = URLEncoder.encode("yzbzz://5?userId=3&phone=186xxx&t=张三", "utf-8")
        httpUrl = URLEncoder.encode("http://www.baidu.com?userId=3&phone=186xxx&t=张三", "utf-8")
    }

    override fun getLayoutId(): Int {
        return R.layout.fragment_work_state
    }

    override fun initView() {
        btn_ok.setOnClickListener {
            if (login.equals("1", ignoreCase = true)) {
                login = "0"
                ToastUtils.showToast("登录关闭")
            } else {
                login = "1"
                ToastUtils.showToast("登录开启")
            }
        }
        btn_yzbzz.setOnClickListener {
            schemeUrl = "yzbzz://3?login=$login"
        }
        btn_http.setOnClickListener {
            schemeUrl = "http://www.baidu.com"
        }
        btn_yzbzz_s_e.setOnClickListener {
            schemeUrl = "yzbzz://100?url=$yzbzzUrl&login=$login"
        }
        btn_yzbzz_s_h.setOnClickListener {
            schemeUrl = "yzbzz://100?url=$httpUrl&login=$login"
        }
        btn_yzbzz_s_d_e.setOnClickListener {
            defaultData = "yzbzz://2"
            schemeUrl = "yzbzz://101?login=$login"
        }
        btn_yzbzz_s_d_h.setOnClickListener {
            defaultData = "http://www.163.com"
            schemeUrl = "yzbzz://101?login=$login"
        }
        btn_yzbzz_s_d_error.setOnClickListener {
            defaultData = "yzbzz://102"
            schemeUrl = "yzbzz://101?login=$login"
        }

        btn_yzbzz_s_d_error_o.setOnClickListener {
            defaultData = "abc**102"
            schemeUrl = "yzbzz://101?login=$login"
        }

    }

    fun startTo(url: String?) {
        val intent = Intent(ctx, TestActivity::class.java)
        val uri = Uri.parse(url)
        intent.data = uri
        intent.putExtra("defaultData", defaultData)
        startActivity(intent)
    }

    companion object {

        fun newInstance(): FragmentA {
            val fragment = FragmentA()
            val args = Bundle()
            fragment.arguments = args
            return fragment
        }
    }

    fun appendUrl(url: String?, urlParams: Map<String, String>?): String {
        if (null == url) {
            return ""
        }
        if (urlParams == null || urlParams.size <= 0) {
            return url
        }
        val httpUrl = url!!.toHttpUrlOrNull() ?: return url
        val urlBuilder = httpUrl.newBuilder()
        for ((key, value) in urlParams) {
            urlBuilder.addQueryParameter(key, value)
        }
        return urlBuilder.build().toString()
    }
}
